src/fides/api/db/crud.py

```diff
@@ -324,6 +327,19 @@ async def delete_resource(
                     log.debug("Deleting resource")
                     query = _delete(sql_model).where(sql_model.fides_key == fides_key)
                 await async_session.execute(query)
+            except IntegrityError as err:
```

Comment:
i know we don't need to listen to everything codecov says, but it feels like we could have a test case to cover this, i.e. i don't think it should be too bad to provoke the error state in a test? specifically - the string matching in the error text may be a good piece of logic to ensure we've got coverage for.

if it's not an easy situation to get test coverage on for whatever reason then not a big deal to me.

Reply:
You're totally right here. I'm deep in some Pydantic stuff but will get this added before merging :)

Reply:
got coverage for this in https://github.com/ethyca/fides/pull/4406/commits/8512d52fae60b5ac2c9e60bba80661a06ae2579c to help get this across the line! i think we're good to go, i'll let CI run through again...